launchable/test_runners/go_test.py

```diff
@@ -1,18 +1,37 @@
 from . import launchable
+import re
+from ..utils.logger import Logger
 
 
 @launchable.subset
 def subset(client):
-    for case in client.stdin():
-        # Avoid last line such as `ok      github.com/launchableinc/rocket-car-gotest      0.268s`
-        if not ' ' in case:
-            client.test_path([{'type': 'testcase', 'name': case.rstrip('\n')}])
+    logger = Logger()
 
-    client.formatter = lambda x: "^{}$".format(x[0]['name'])
+    # NOTE: This should be using package name + test function name to specify
+    # which tests to run. However, the initial integration is created so that we
+    # only specify the test function names. This can result in matching some
+    # extra tests in multiple packages. However, in order to keep the initial
+    # way of the integration, we cannot change this. Try to do the best.
+    test_cases = []
+    for line in client.stdin():
+        if not ' ' in line:
+            test_cases.append(line.strip('\n'))
+        else:
+            parts = re.split('\s+', line)
+            if len(parts) >= 2:
+                package = parts[1].split('/')[-1]
```

Comment:
I checked this fixing used by https://github.com/launchableinc/examples/tree/master/go

Then, test report produces class name as an absolute  e.g) `"github.com/launchableinc/examples/go/example"`, `"github.com/launchableinc/examples/go/example"`

<details>
`$ launchable record tests go-test ./report.xml`


```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="6">
        <testsuite name="github.com/launchableinc/examples/go" tests="4" failures="0" errors="0" id="0" hostname="RyosukeYabukis-MacBook-Pro.local" time="5.000" timestamp="2022-08-24T09:07:47+09:00">
                <testcase name="TestExample1" classname="github.com/launchableinc/examples/go" time="0.000"></testcase>
                <testcase name="TestExample2" classname="github.com/launchableinc/examples/go" time="3.000"></testcase>
                <testcase name="TestExample3" classname="github.com/launchableinc/examples/go" time="0.000"></testcase>
                <testcase name="TestExample4" classname="github.com/launchableinc/examples/go" time="2.000"></testcase>
        </testsuite>
        <testsuite name="github.com/launchableinc/examples/go/example" tests="2" failures="0" errors="0" id="1" hostname="RyosukeYabukis-MacBook-Pro.local" time="0.000" timestamp="2022-08-24T09:07:47+09:00">
                <testcase name="TestGreeting" classname="github.com/launchableinc/examples/go/example" time="0.000"></testcase>
                <testcase name="ExampleGreeting" classname="github.com/launchableinc/examples/go/example" time="0.000"></testcase>
        </testsuite>
</testsuites>
```

</details>

However, this subset uses only last name of package e.g) ` [{'type': 'class', 'name': 'example'}, {'type': 'testcase', 'name': 'TestGreeting'}]`

I have a question, can the Launchable model evaluate use by only the last name of a package?

Reply:
Oh... go-junit-report produces a different class format between v1 and v2...

```
$ go-junit-report --version
go-junit-report v1.0.0-dev  (HEAD)

$ go test -v ./... | ~/bin/go-junit-report
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
        <testsuite tests="4" failures="0" time="0.000" name="github.com/launchableinc/examples/go">
                <properties>
                        <property name="go.version" value="go1.19"></property>
                </properties>
                <testcase classname="go" name="TestExample1" time="0.000"></testcase>
                <testcase classname="go" name="TestExample2" time="3.000"></testcase>
                <testcase classname="go" name="TestExample3" time="0.000"></testcase>
                <testcase classname="go" name="TestExample4" time="2.000"></testcase>
        </testsuite>
        <testsuite tests="2" failures="0" time="0.000" name="github.com/launchableinc/examples/go/example">
                <properties>
                        <property name="go.version" value="go1.19"></property>
                </properties>
                <testcase classname="example" name="TestGreeting" time="0.000"></testcase>
                <testcase classname="example" name="ExampleGreeting" time="0.000"></testcase>
        </testsuite>
</testsuites>
```

```
$ go-junit-report --version
go-junit-report v2.0.0-dev  (HEAD)

$ go test -v ./... | ~/bin/go-junit-report
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="6">
        <testsuite name="github.com/launchableinc/examples/go" tests="4" failures="0" errors="0" id="0" hostname="RyosukeYabukis-MacBook-Pro.local" time="5.000" timestamp="2022-08-24T11:23:46+09:00">
                <testcase name="TestExample1" classname="github.com/launchableinc/examples/go" time="0.000"></testcase>
                <testcase name="TestExample2" classname="github.com/launchableinc/examples/go" time="3.000"></testcase>
                <testcase name="TestExample3" classname="github.com/launchableinc/examples/go" time="0.000"></testcase>
                <testcase name="TestExample4" classname="github.com/launchableinc/examples/go" time="2.000"></testcase>
        </testsuite>
        <testsuite name="github.com/launchableinc/examples/go/example" tests="2" failures="0" errors="0" id="1" hostname="RyosukeYabukis-MacBook-Pro.local" time="0.000" timestamp="2022-08-24T11:23:46+09:00">
                <testcase name="TestGreeting" classname="github.com/launchableinc/examples/go/example" time="0.000"></testcase>
                <testcase name="ExampleGreeting" classname="github.com/launchableinc/examples/go/example" time="0.000"></testcase>
        </testsuite>
</testsuites>
```

Reply:
Which version should we support as default...??